### PR TITLE
Opt-in omit x-ms-external model generation

### LINKED ISSUE #876: Opt-in `x-ms-external` behavior change
Recent change of this [PR](https://github.com/Azure/autorest.go/pull/862) causes breaking changes of mgmt. modules. Need to fix with opt-in.

### PR DESCRIPTION
**Pull request by tadelesh**
Resolve #876

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Can you please provide some examples of how this is breaking existing modules?

**Comment by tadelesh** [comment]
> Can you please provide some examples of how this is breaking existing modules?

@jhendrixMSFT Some recent releases of mgmt. modules got breaking changes by this:

kusto: https://github.com/Azure/azure-sdk-for-go/pull/19124, [swagger](https://github.com/Azure/azure-rest-api-specs/blob/f70b7953f5d54889cc1825b37ad938342ca93a2e/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2022-07-07/kusto.json#L5636)
netapp: https://github.com/Azure/azure-sdk-for-go/pull/19132, [swagger](https://github.com/Azure/azure-rest-api-specs/blob/0507c3d410681049bb5cd9545b49a7578a91d3d0/specification/netapp/resource-manager/Microsoft.NetApp/stable/2022-05-01/netapp.json#L7291)
containerservice: https://github.com/Azure/azure-sdk-for-go/pull/19111, [swagger](https://github.com/Azure/azure-rest-api-specs/blob/2a07629a9919989375ab04490fb24051f78d1a7c/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json#L4787)

**Comment by tadelesh** [comment]
@jhendrixMSFT Many releases are blocked by this. Could you help to review ASAP. Thanks.

**Comment by jhendrixMSFT** [comment]
Can we not work around these by way of transformation?  Or are there more than just three releases?

File: src/generator/models.ts [hunk 1]
```diff
@@ -25,7 +25,8 @@ export async function generateModels(session: Session<CodeModel>): Promise<model
   let modelText = await contentPreamble(session);
 
   // we do model generation first as it can add imports to the imports list
-  const structs = generateStructs(modelImports, serdeImports, session.model.schemas.objects);
+  const omitXMsExternal = await session.getValue('omit-x-ms-external', true);
```
Comment (jhendrixMSFT) [inline src/generator/models.ts:28]: I would prefer that the default is `false`, and that if `azure-arm` is `true` then we can default this to `true`.

**Comment by tadelesh** [comment]
@jhendrixMSFT I think this change could be closed as Jeff said the removal unreferenced type could be treated as non-breaking change ([link](https://github.com/Azure/azure-sdk-for-go/pull/19124#issuecomment-1251365204)).

**Pull request closed without merging**